tools/image-info

```diff
@@ -2488,161 +2476,208 @@ def ensure_device_file(path: str, major: int, minor: int):
+    }
 
-    finally:
-        r = subprocess.run(["vgchange", "-an", vg_name],
-                            stdout=subprocess.DEVNULL,
-                            stderr=subprocess.PIPE,
-                            check=False)
-        if r.returncode != 0:
-            raise RuntimeError(r.stderr.strip())
 
 
 def partition_is_lvm(part: Dict) -> bool:
     return part["type"].upper() in ["E6D6D379-F507-44C2-A23C-238F2A3DF928", "8E"]
 
 
-def append_partitions(report, image, loctl):
+def append_partitions(report, image, devpath:str, cm: contextlib.ExitStack, mgr:host.ServiceManager):
     partitions = report["partitions"]
 
-    with contextlib.ExitStack() as cm:
-        # open each partition as a loop device
-        filesystems = {}
-        for part in partitions:
-            start, size = part["start"], part["size"]
-            dev = cm.enter_context(loop_open(loctl, image, offset=start, size=size))
-            read_partition(dev, part)
-            if partition_is_lvm(part):
-                lvm = cm.enter_context(discover_lvm(dev))
-                for vol in lvm["lvm.volumes"].values():
-                    if vol["fstype"]:
-                        mntopts = []
-                        # we cannot recover since the underlying loopback device is mounted
-                        # read-only but since we are using the it through the device mapper
-                        # the fact might not be communicated and the kernel attempt a to
-                        # a recovery of the filesystem, which will lead to a kernel panic
-                        if vol["fstype"] in ("ext4", "ext3", "xfs"):
-                            mntopts = ["norecovery"]
-                        filesystems[vol["uuid"].upper()] = {
-                            "device": vol["device"],
-                            "mntops": mntopts
-                        }
-                    del vol["device"]
-                part.update(lvm)
-            elif part["uuid"] and part["fstype"]:
-                filesystems[part["uuid"].upper()] = {
-                    "device": dev
-                }
+    devmgr = devices.DeviceManager(mgr, devpath, os.path.dirname(image))
+    devices_map = {}
+    filesystems = {}
+    # open each partition as a loop device
+    for part in partitions:
+        start, size = part["start"], part["size"]
+        ret = cm.enter_context(
+            loop_open(
+                devpath,
+                devmgr,
+                part["partuuid"],
+                image,
+                size,
+                offset=start)
+        )
+        devices_map[ret["path"]] = ret["dev"]
+        read_partition(ret["path"], part)
+        if partition_is_lvm(part):
+            # Temporary hack:
+            # It seems that for some distros pvdisplay doesn't manage to open
+            # the device when we're working under with /dev in the temporary
+            # directory.
+            tst = f"/dev{ret['path'].split('dev')[1]}"
```

Comment:
So this is part of the shady things. I wanted to stop loopback mounting in `/dev` at some point and to bind mount under `/tmpt/somethinh/dev'. So I created a tmpfs to do this, created the devpath pointing to the tmpfs and everything was working great! Except for these two distributions centos9 and rhel91 that were giving me a headache! Because on those two, and on those two only, I could not explore the file under /tmp, like I would endup having errors with lvdisplay like `Could not find parent device`.

So I implemented this hack, that is quite ugly but, functionnal...

The question I'm wondering now is, is it useful to have this `tmpfs` after all ?
So I'll will make some tests without, without the hack and see what's what.
If it works well, I'll remove the hack and the tmpfs thing.